Makefile

```diff
@@ -5,6 +5,39 @@ DOCKER_BUF := $(DOCKER) run --rm -v $(CURDIR):/workspace --workdir /workspace bu
 IMAGE := ghcr.io/tendermint/docker-build-proto:latest
 DOCKER_PROTO_BUILDER := docker run -v $(shell pwd):/workspace --workdir /workspace $(IMAGE)
 PROJECTNAME=$(shell basename "$(PWD)")
+LEDGER_ENABLED ?= true
+
+
+# process build tags
+build_tags =
+ifeq ($(LEDGER_ENABLED),true)
+  ifeq ($(OS),Windows_NT)
+    GCCEXE = $(shell where gcc.exe 2> NUL)
+    ifeq ($(GCCEXE),)
+      $(error gcc.exe not installed for ledger support, please install or set LEDGER_ENABLED=false)
+    else
+      build_tags += ledger
+    endif
+  else
+    UNAME_S = $(shell uname -s)
+    ifeq ($(UNAME_S),OpenBSD)
+      $(warning OpenBSD detected, disabling ledger support (https://github.com/cosmos/cosmos-sdk/issues/1988))
+    else
+      GCC = $(shell command -v gcc 2> /dev/null)
+      ifeq ($(GCC),)
+        $(error gcc not installed for ledger support, please install or set LEDGER_ENABLED=false)
+      else
+        build_tags += ledger
+      endif
+    endif
+  endif
+endif
+
```

Comment:
[blocking] I think this is missing a command:

```
build_tags := $(strip $(build_tags))
```

because I added this to the Makefile:

```Makefile
## debug: Print debug info.
debug:
	@echo build_tags_comma_sep: $(build_tags_comma_sep)
	@echo build_tags: $(build_tags)
	@echo ldflags: $(ldflags)
	@echo BUILD_FLAGS: $(BUILD_FLAGS)
```

Output

```
$ make debug
build_tags_comma_sep: ,ledger
build_tags: ledger
ldflags: -X github.com/cosmos/cosmos-sdk/version.Name=celestia-app -X github.com/cosmos/cosmos-sdk/version.AppName=celestia-appd -X github.com/cosmos/cosmos-sdk/version.Version=1.0.0-rc0-90-ga668ce67 -X github.com/cosmos/cosmos-sdk/version.Commit=a668ce67ed393cb2a11f3d53883d9331d6034d04 -X github.com/cosmos/cosmos-sdk/version.BuildTags=,ledger
BUILD_FLAGS: -tags  ledger -ldflags -X github.com/cosmos/cosmos-sdk/version.Name=celestia-app -X github.com/cosmos/cosmos-sdk/version.AppName=celestia-appd -X github.com/cosmos/cosmos-sdk/version.Version=1.0.0-rc0-90-ga668ce67 -X github.com/cosmos/cosmos-sdk/version.Commit=a668ce67ed393cb2a11f3d53883d9331d6034d04 -X "github.com/cosmos/cosmos-sdk/version.BuildTags=,ledger"
```

Note that there is a leading `,` before ledger. This doesn't appear to be the case for Osmosis:

```
$ make debug
build_tags_comma_sep: netgo,ledger
build_tags: netgo ledger
ldflags: -X github.com/cosmos/cosmos-sdk/version.Name=osmosis -X github.com/cosmos/cosmos-sdk/version.AppName=osmosisd -X github.com/cosmos/cosmos-sdk/version.Version=crosschain-swaps-v1.0.0-685-gfaacd4efe -X github.com/cosmos/cosmos-sdk/version.Commit=faacd4efebcf3142c9af1abcf6cf4ad377e32d40 -X github.com/cosmos/cosmos-sdk/version.BuildTags=netgo,ledger -w -s
BUILD_FLAGS: -tags netgo ledger -ldflags -X github.com/cosmos/cosmos-sdk/version.Name=osmosis -X github.com/cosmos/cosmos-sdk/version.AppName=osmosisd -X github.com/cosmos/cosmos-sdk/version.Version=crosschain-swaps-v1.0.0-685-gfaacd4efe -X github.com/cosmos/cosmos-sdk/version.Commit=faacd4efebcf3142c9af1abcf6cf4ad377e32d40 -X "github.com/cosmos/cosmos-sdk/version.BuildTags=netgo,ledger" -w -s -trimpath
```

If we add 

```suggestion
build_tags := $(strip $(build_tags))
```

then the output doesn't contain a leading `,`

```
$ make debug
build_tags_comma_sep: ledger
build_tags: ledger
ldflags: -X github.com/cosmos/cosmos-sdk/version.Name=celestia-app -X github.com/cosmos/cosmos-sdk/version.AppName=celestia-appd -X github.com/cosmos/cosmos-sdk/version.Version=1.0.0-rc0-90-ga668ce67 -X github.com/cosmos/cosmos-sdk/version.Commit=a668ce67ed393cb2a11f3d53883d9331d6034d04 -X github.com/cosmos/cosmos-sdk/version.BuildTags=ledger
BUILD_FLAGS: -tags ledger -ldflags -X github.com/cosmos/cosmos-sdk/version.Name=celestia-app -X github.com/cosmos/cosmos-sdk/version.AppName=celestia-appd -X github.com/cosmos/cosmos-sdk/version.Version=1.0.0-rc0-90-ga668ce67 -X github.com/cosmos/cosmos-sdk/version.Commit=a668ce67ed393cb2a11f3d53883d9331d6034d04 -X "github.com/cosmos/cosmos-sdk/version.BuildTags=ledger"
```

`strip` appears to remove leading / trailing whitespace.

Ref: https://www.gnu.org/software/make/manual/html_node/Text-Functions.html

Reply:
Great catch!